File: apps/mobile/lib/config/tenant/tenant_modules.dart
/// Compile-time business module toggles — injected via --dart-define-from-file.
///
/// Unlike FeatureFlags (generic UX preferences, same for every tenant), these
/// are tied to the tenant's business domain: seeded from TenantType at
/// creation on the backend (see Identity's TenantModules.CreateDefaults), and
/// drive which tabs exist in the bottom navigation (see bottom_nav_tabs.dart).
///
/// Defaults to true so every module is visible during local development even
/// if the .env file is not provided — every real build's checked-in .env/*.json
/// always sets these explicitly, so the default only matters for ad-hoc
/// `flutter run` without --dart-define-from-file.
class TenantModules {
  static final TenantModules _instance = TenantModules._internal();
  factory TenantModules() => _instance;
  TenantModules._internal();

  final bool events =
      const bool.fromEnvironment('MODULE_EVENTS', defaultValue: true);

  final bool inventory =
      const bool.fromEnvironment('MODULE_INVENTORY', defaultValue: true);

  final bool clients =
      const bool.fromEnvironment('MODULE_CLIENTS', defaultValue: true);
}

File: apps/backend/services/Identity/Identity.Domain/Entities/TenantModules.cs
using Identity.Domain.Enums;

namespace Identity.Domain.Entities;

/// <summary>
/// Which business modules/pages this tenant has access to — drives the mobile app's
/// bottom navigation. Unlike <see cref="FeatureFlags"/> (generic UX preferences), these
/// are tied to the tenant's business domain and seeded from <see cref="TenantType"/> at
/// creation time; each can still be toggled independently afterwards.
///
/// Only covers modules that actually exist today (Rental's Events/Inventory/Clients,
/// backed by the Business service). Reporting and Poultry are deliberately not modeled
/// yet — add them here once those modules are actually built.
/// </summary>
public class TenantModules
{
    public bool Events { get; set; }
    public bool Inventory { get; set; }
    public bool Clients { get; set; }

    public static TenantModules CreateDefaults(TenantType type) => type switch
    {
        TenantType.Rental => new TenantModules { Events = true, Inventory = true, Clients = true },
        TenantType.Farming => new TenantModules { Inventory = true },
        _ => new TenantModules()
    };
}
